### PR TITLE
tests/float: automated test and added pexpect script

### PR DESCRIPTION
**Pull request by haukepetersen**
One more done...

### CODE REVIEW

**Comment by haukepetersen** [comment]
No sure I understand. But for what reason would good would do the output of all the plus and minuses, except that they make the test run slower (printf overhead...)? And what sense does it make to continue the test after it has failed once?

**Comment by miri64** [comment]
> But for what reason would good would do the output of all the plus and minuses, except that they make the test run slower (printf overhead...)?

Keeping the test intact as is + if you output also positive results you can move the iteration to the test script.

> And what sense does it make to continue the test after it has failed once?

It doesn't... The test stops when the exception is raised.

**Comment by haukepetersen** [comment]
> Keeping the test intact as is + if you output also positive results you can move the iteration to the test script

Ah, you mean that you keep the number of iterations out of the test application all together and simply parameterize it in the test script?

**Comment by haukepetersen** [comment]
Ok, played with that a little bit: going with your proposed solution makes a tremendous difference in run-time: doing 10 million iterations on `natvie` with printing `-` signs takes ~2min. Doing the same test with this PR takes ~10ms... In my opinion this speaks much towards this PR!

**Comment by miri64** [comment]
(but Murdock has an issue with this PR)

**Comment by haukepetersen** [comment]
Murdock should be happy now.

File: tests/float/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# directory for more details.
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+def testfunc(child):
+    child.expect(u"Testing floating point arithmetics \(\d+ iterations\)...")
```
Comment (miri64) [inline tests/float/tests/01-run.py]: This fails now (is this line [in the c-code also] necessary at all).

Comment (haukepetersen) [reply tests/float/tests/01-run.py]: it prevents from accidentally parsing the output of a previous run (output by board before calling `make reset`), so I think it makes sense to keep it.

**Comment by haukepetersen** [comment]
fixed pexpect script

**Comment by miri64** [comment]
Please provide a backport to 2017.10

**Comment by haukepetersen** [comment]
backport provided in #7835

**Pull request merged**